scheduler/mesos/mesos.go

```diff
@@ -0,0 +1,88 @@
+package mesos
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/docker/swarm/cluster"
+	"github.com/docker/swarm/discovery"
+	"github.com/docker/swarm/scheduler/options"
+	"github.com/samalba/dockerclient"
+)
+
+var ErrNotImplemented = errors.New("not implemented in the mesos scheduler")
+
+type MesosScheduler struct {
+	sync.Mutex
+
+	//TODO: list of mesos masters
+	cluster *cluster.Cluster
+	options *options.SchedulerOptions
+}
+
+func (s *MesosScheduler) Initialize(options *options.SchedulerOptions) {
+	s.options = options
+
+	s.cluster = cluster.NewCluster(s.options.Store)
+}
+
+// Schedule a brand new container into the cluster.
+func (s *MesosScheduler) CreateContainer(config *dockerclient.ContainerConfig, name string) (*cluster.Container, error) {
+
+	s.Lock()
+	defer s.Unlock()
+
+	//TODO: RequestOffers from mesos master
+
+	//TODO: pick the right offer (using strategy & filters ???)
+
+	//TODO: LaunchTask on the Mesos cluster and get container
+
+	//TODO: Store container in store
+	//  -   s.cluster.CommitContainerInStore(container.Id, config, name)
+
+	return nil, ErrNotImplemented
+}
+
+// Remove a container from the cluster. Containers should always be destroyed
+// through the scheduler to guarantee atomicity.
+func (s *MesosScheduler) RemoveContainer(container *cluster.Container, force bool) error {
+	s.Lock()
+	defer s.Unlock()
+
+	//TODO: KillTask
+
+	//TODO: remove container
+	//  -   s.cluster.RemoveContainerFromStore(container.Id, force)
+
+	return ErrNotImplemented
+}
+
+// Entries are Mesos masters
+func (s *MesosScheduler) NewEntries(entries []*discovery.Entry) {
```

Comment:
When you say receive Mesos masters, are you referring when the master changes? 
Swarm definitely needs the list of nodes, and you can query the master for all the slaves. Looks like this is being called back from discovery service. One concern I have is that since this is a seperate discovery process that is proxying information, if some reason the mesos slave is down it will still return information given the node is still alive.
If all the docker commands that it needs to proxy is about a running container, it is much better to get the information from the mesos as that's more definitive.